rlp/rlpgen/main.go

```diff
@@ -52,8 +54,15 @@ func main() {
 	}
 	if *output == "-" {
 		os.Stdout.Write(code)
-	} else if err := ioutil.WriteFile(*output, code, 0644); err != nil {
-		fatal(err)
+	} else {
+		canonicalPath, err := common.VerifyPath(*output)
```

Comment:
can we write a common code for this in a helper for to verifypath and then write file?
maybe instead of ioutil.writefile/readfile we can use a helper function and there we can do - `util.readfile`/`util.writefile`
seeing similar changes at a couple of places

Reply:
I'd like to keep the security related path verification separated from the action performed with the `canonicalPath` executed afterwards. That function may be file `write` or `read`, directory `mkdir` or even an `extractArchive`. Imho having some separation of concern here could be helpful in identifying potential issues with the path, and also pointing out security related issues before read/write operations. What do you think?